src/main.rs

```diff
@@ -291,6 +297,8 @@ fn main() {
 
     #[cfg(unix)]
     {
+        use future::Either;
```

Comment:
one last nit: I think all these Either is very hard to read. I wonder if we can refactor both unix and windows impls to use a custom struct then move the actual handling of each type of signal in a common place?

Reply:
Yes, I was thinking the same thing, but didn't want to change main any more than needed. 
It's possible, move signal streams to a stream struct, unify the signals to a single command enum, like 
```
enum SignalEvent{
 Shutdown,
 ShutdownImmediately,
 ReloadConfig,
} 
```
, and yes handling of them could be unified.

Then all I add this.